bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/TelescopePosTableWidget.java

```diff
@@ -922,31 +922,27 @@ void updatePrimaryStar() {
         final Option<IndexedGuideGroup> iggOpt = targetOpt.flatMap(this::getTargetGroup).orElse(getSelectedGroup());
         final boolean primaryGroupIsSelected = iggOpt.exists(igg -> igg.group().equals(env.getPrimaryGuideGroup()));
 
-        if (!primaryGroupIsSelected) {
-            // The current target or guide group is not primary, so mark ths owner guide group as primary.
-            iggOpt.foreach(igg -> {
-                final GuideGroup primary = env.getPrimaryGuideGroup();
-
-                // If the auto group is disabled and set to primary, then make it an initial auto group.
-                ImOption.apply(env.getGuideEnvironment()).foreach(ge -> {
-                    final GuideGrp grp = igg.group().grp();
-                    if (ge != null && grp instanceof AutomaticGroup.Disabled$) {
-                        final TargetEnvironment envNew = env.setGuideEnvironment(ge.setAutomaticGroup(GuideGroup.AutomaticInitial()).setPrimaryIndex(0));
-                        _dataObject.setTargetEnvironment(envNew);
-                        _model.enableAutoRow(envNew);
+        iggOpt.foreach(igg -> {
+            final GuideGroup primary = env.getPrimaryGuideGroup();
+
+            // If the auto group is disabled and set to primary, then make it an initial auto group.
+            ImOption.apply(env.getGuideEnvironment()).foreach(ge -> {
+                final GuideGrp grp = igg.group().grp();
+                if (ge != null && grp instanceof AutomaticGroup.Disabled$) {
```

Comment:
You are right. If `env.getGuideEnvironment` was null, the `ImOption.apply` call would turn it into `None`, effectively guaranteeing that `ge` can never be null.

I'll take this part of the `if` clause out.